shell/browser/api/electron_api_push_notifications_mac.mm

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022 Asana, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/api/electron_api_push_notifications.h"
+
+#include <string>
+
+#include <utility>
+#include <vector>
+#import "shell/browser/mac/electron_application.h"
+#include "shell/common/gin_converters/value_converter.h"
+#include "shell/common/gin_helper/promise.h"
+
+namespace electron {
+
+namespace api {
+
+// This set maintains all the promises that should be fulfilled
+// once macOS registers, or fails to register, for APNS
+std::vector<gin_helper::Promise<std::string>> apns_promise_set_;
```

Comment:
ah missed this, this should be moved inside the class please!

Reply:
Updated!